btk/metrics.py

```diff
@@ -1,484 +1,365 @@
-        batch_index(int): Index number of the batch.
-    Returns:
-        batch_true_table: astropy.table.Table with parameters of true galaxy in
-            the batch.
-        batch_detected_table: astropy.table.Table with parameters of detected
-            objects in the batch.
-        batch_blend_list: List summarizing detection match results.
-    """
-    batch_true_table = astropy.table.Table()
-    batch_detected_table = astropy.table.Table()
-    batch_size = len(true_tables)
-    batch_blend_summary = []
-    for i in range(batch_size):
-        true_table = true_tables[i]
-        detected_table = detected_tables[i]
-        # initialize columns to default values
-        initialize_detection_tables(detected_table, true_table, batch_index, batch_size, i)
-        # match detection and true source
-        get_detection_match(true_table, detected_table)
-        # summarize blend detection results to table
-        blend_summary = get_blend_detection_summary(true_table, detected_table)
-        batch_blend_summary.append(blend_summary)
-        # add results to batch table
-        batch_detected_table = astropy.table.vstack((batch_detected_table, detected_table))
-        batch_true_table = astropy.table.vstack((batch_true_table, true_table))
-    return batch_true_table, batch_detected_table, batch_blend_summary
-
-
-def evaluate_segmentation(segmentation, data=None, index=None):
-    if segmentation is None:
-        return None
-    return None
-
-
-def evaluate_flux(flux, data=None, index=None):
-    if flux is None:
-        return None
-    return None
-
-
-def evaluate_shapes(shapes, data=None, index=None):
-    if shapes is None:
-        return None
-    return None
-
-
-def run(metrics_params, test_size=1000):
-    """Runs detection/segmentation/flux/shape measurement algorithm defined in
-    the input metrics params for input test_size number of btk runs.
+    results_detection = {}
+    precision = []
+    recall = []
+    f1 = []
+    for i in range(len(blend_list)):
+        matches_blend = matches[i]["match_detected_id"]
+        true_pos = 0
+        false_pos = 0
+        false_neg = 0
+        for match in matches_blend:
+            if match == -1:
+                false_neg += 1
+            else:
+                true_pos += 1
+        for j in range(len(detection_catalogs[i])):
+            if j not in matches_blend:
+                false_pos += 1
+        if true_pos + false_pos > 0:
+            precision.append(true_pos / (true_pos + false_pos))
+        else:
+            precision.append(0)
+        recall.append(true_pos / (true_pos + false_neg))
+        if precision[-1] != 0 and recall[-1] != 0:
+            f1.append(2 / (1 / precision[-1] + 1 / recall[-1]))
+        else:
+            f1.append(0)
+    results_detection["precision"] = precision
+    results_detection["recall"] = recall
+    results_detection["f1"] = recall
+    return results_detection
```

Comment:
This values don't get added to the `galaxy_summary` table later because they are 'per blend' quantities right? 

Two things, (1) Would it be a good idea to have another table `blend_summary` with information for each blend? I can image this might simplify things for some users looking to summarize their results. (2) Is it useful to calculate these 3 quantities for each blend or should we just do it once for some large quantity of batches? Is the average `f1` over batches the same as calculating `f1` for all batches combined as a single dataset?

Reply:
Those are per blend quantities, but I thought some more and they are probably better to calculate on batch, and I don't believe you can just average them afterward ; I will get on it later.